lib/matplotlib/backend_managers.py

```diff
@@ -14,6 +18,180 @@
 from matplotlib.rcsetup import validate_stringlist
 import matplotlib.backend_tools as tools
 
+from matplotlib import is_interactive
+from matplotlib import rcParams
+from matplotlib.figure import Figure
+from matplotlib.backend_bases import key_press_handler
+from matplotlib.backends import get_backend
+
+
+class FigureManagerEvent(object):
+    """Event for when something happens to this figure manager.
+    i.e. the figure it controls gets closed
+
+    Attributes
+    ----------
+    signal : str
+        The name of the signal.
+
+    figure_manager : FigureManager
+        The figure manager that fired the event.
+    """
+    def __init__(self, signal, figure_manager):
+        self.name = signal
+        self.figure_manager = figure_manager
+
+
+class FigureManager(cbook.EventEmitter):
+    """
+    The FigureManager creates and wraps the necessary components to display a
+    figure, namely the Window, FigureCanvas and Toolbar.  It gets used whenever
+    you want the figure in a standalone window.
+
+    Parameters
+    ----------
+    figure : `matplotlib.figure.Figure`
+        The figure to manage.
+
+    num : int
+        The figure number.
+
+    Attributes
+    ----------
+
+    canvas : `matplotlib.backend_bases.FigureCanvasBase`
+        The GUI element on which we draw.
+
+    toolbar : `matplotlib.backend_bases.NavigationToolbar2`
+        The toolbar used for interacting with the figure.
+
+    window : `matplotlib.backend_bases.WindowBase`
+        The window that holds the canvas and toolbar.
+
+    num : int
+        The figure number.
+    """
+    def __init__(self, figure, num):
```

Comment:
Because we want to create a ``FigureManager`` not a ``CanvasManager`` right?

Also much cleaner, allowing us to get rid of the ``new_figure_manager`` and ``new_figure_manager_from_figure`` ugly methods.